lib/ansible/modules/cloud/docker/docker_stack.py

```diff
@@ -261,8 +261,8 @@ def main():
 
         if rc != 0:
             module.fail_json(msg="docker stack up deploy command failed",
-                             out=out,
-                             rc=rc, err=err)
+                             stdout=out,
```

Comment:
While the other changes are ok, this one is problematic. If a playbook or role depends on the currently named return values, it will stop working. It's better to return all four (`out`, `stdout`, `err`, `stderr`) for now, and announce in the docs and porting guide that `out` and `err` are deprecated and will be removed in Ansible 2.14 (standard deprecation cycle of four versions).

Reply:
Full enclosure: I know very little about what is done with the values returned from fail_json.
As you can see in my example result from before the changes, only `changed` and `failed` were exposed to me as an ansible user. This is what prompted my changes to begin with.
My point is that I'm not sure anyone could rely on the existing return values, if they were not exposed.

Reply:
@parmsib 
Your example is for a succesful module run: `out`, `rc` and `stderr` were included only on module failure (https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/docker/docker_stack.py#L263-L265 and https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/docker/docker_stack.py#L289-L292)

Reply:
Ah, thanks for the info.
One more question: Why is specifically the changes to line 264 bad, and e.g not [283](https://github.com/ansible/ansible/blob/48af1b90ea3558b1cbe7512d587481f945b50025/lib/ansible/modules/cloud/docker/docker_stack.py#L283)? Is it because the latter is exiting, and therefore we don't expect a user to be reading the values programmatically?

Reply:
Because it renames an existing return value, instead of adding a new one. If some existing role or playbook depends on the return value being named `out` resp. `err` in case of this error, it will break. Ansible tries to avoid such breakage in minor versions.

Reply:
> While the other changes are ok, this one is problematic. If a playbook or role depends on the currently named return values, it will stop working. It's better to return all four (`out`, `stdout`, `err`, `stderr`) for now, and announce in the docs and porting guide that `out` and `err` are deprecated and will be removed in Ansible 2.14 (standard deprecation cycle of four versions).

I may need some guidance on what exactly to do here. Which porting guide page do I add this to, and is there somewhere I should add a "reminder" about deprecating the old return values?

Reply:
Mention this in the changelog (`deprecated_features` section) and in the [porting guide](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guides/porting_guide_2.10.rst#deprecation-notices). And also in the module's documentation.